Content.Server/CartridgeLoader/Cartridges/CrewManifestCartridgeSystem.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Content.Server.CrewManifest;
+using Content.Server.Station.Systems;
+using Content.Shared.CartridgeLoader;
+using Content.Shared.CartridgeLoader.Cartridges;
+using Content.Shared.CCVar;
+using Robust.Shared.Configuration;
+
+namespace Content.Server.CartridgeLoader.Cartridges;
+
+public sealed class CrewManifestCartridgeSystem : EntitySystem
+{
+
+    [Dependency] private readonly CartridgeLoaderSystem? _cartridgeLoader = default!;
+    [Dependency] private readonly StationSystem _stationSystem = default!;
+    [Dependency] private readonly CrewManifestSystem _crewManifest = default!;
+    [Dependency] private readonly IConfigurationManager _configManager = default!;
+
+    private const string CartridgePrototypeName = "CrewManifestCartridge";
+
+    private bool _allowed = true;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<CrewManifestCartridgeComponent, CartridgeMessageEvent>(OnUiMessage);
+        SubscribeLocalEvent<CrewManifestCartridgeComponent, CartridgeUiReadyEvent>(OnUiReady);
+        SubscribeLocalEvent<ProgramInstallationAttempt>(OnInstallationAttempt);
+        _configManager.OnValueChanged(CCVars.CrewManifestUnsecure, OnCrewManifestUnsecureChanged, true);
+    }
```

Comment:
You need to unsubscribe from value changes as C# events are backing them.

Reply:
Sorry, could you please explain a little bit further?

Reply:
So in the engine it will subscribe to this value changing, but if the server shutdowns and re-initializes it will subscribe again and double it, so you should unsubscribe the value changed on shutdown.

Reply:
Added OnShutdown